server/src/main/java/io/druid/segment/realtime/plumber/RealtimePlumber.java

```diff
@@ -468,6 +472,8 @@ public void doRun()
                 // This call will also delete possibly-partially-written files, so we don't need to do it explicitly.
                 abandonSegment(truncatedTime, sink);
               }
+
+              throw Throwables.propagate(e);
```

Comment:
This isn't actually going to fix #1166, since this error will happen in the mergeExecutor and not get reported up. To fix it we need to set some sort of failure flag in addition to calling abandonSegment here, and then check that failure flag in the finishJob method after the sinks become empty.